src/deadline_worker_agent/sessions/actions/run_attachment_upload.py

```diff
@@ -0,0 +1,190 @@
+                embeddedFiles=[
+                    EmbeddedFileText_2023_09(
+                        name="AttachmentUpload",
+                        filename="upload.py",
+                        type=EmbeddedFileTypes_2023_09.TEXT,
+                        data=data,
+                    )
+                ],
+            )
+
+    def __eq__(self, other: Any) -> bool:
+        return (
+            type(self) is type(other)
+            and self._id == other._id
+            and self._step_id == other._step_id
+            and self._task_id == other._task_id
+            and self._step_script == other._step_script
+        )
+
+    def start(
+        self,
+        *,
+        session: Session,
+        executor: Executor,
+    ) -> None:
+        """Initiates job attachments output upload
+
+        Parameters
+        ----------
+        session : Session
+            The Session that is the target of the action
+        executor : Executor
+            An executor for running futures
+        """
+
+        # Banner mimicing the one printed by the openjd-sessions runtime
+        # TODO - Consider a better approach to manage the banner title
+        self._logger.info(
+            "==============================================",
+            extra={"openjd_log_content": LogContent.BANNER},
+        )
+        self._logger.info(
+            "--------- AttachmentUploadAction ---------",
+            extra={"openjd_log_content": LogContent.BANNER},
+        )
+        self._logger.info(
+            "==============================================",
+            extra={"openjd_log_content": LogContent.BANNER},
+        )
+
+        if not (job_attachment_settings := session._job_details.job_attachment_settings):
+            raise RuntimeError("Job attachment settings were not contained in JOB_DETAILS entity")
+
+        assert job_attachment_settings.s3_bucket_name is not None
+        assert job_attachment_settings.root_prefix is not None
+
+        s3_settings = JobAttachmentS3Settings(
+            s3BucketName=job_attachment_settings.s3_bucket_name,
+            rootPrefix=job_attachment_settings.root_prefix,
+        )
+
+        manifest_paths_by_root = session._manifest_paths_by_root
+        output_path = os.path.join(session._session.working_directory, "diff")
+        manifests = list()
+
+        for root, path in manifest_paths_by_root.items():
+            self._logger.info(
+                f"Snapshooting manifest {path} for local root {root}",
+                extra={"openjd_log_content": LogContent.PARAMETER_INFO},
+            )
+            manifest: Optional[ManifestSnapshot] = _manifest_snapshot(
+                root=root,
+                destination=str(output_path),
+                # `output` is used for job download to discover output manifests
+                # manifest file name need to contain the hash of root path for attachment CLI path mapping
+                name=f"output-{os.path.basename(path)}",
```

Comment:
No - because "output" is our own designation for output manifests. For customers, they can use "snapshot" to capture the state of any directory, and can be input or output. 

I won't abstract it into `_manifest_snapshot` directory. If need be, we can make a helper function like `format_output_manfiest_name`